app/controllers/data_explorer/query_controller.rb

```diff
@@ -0,0 +1,236 @@
+class DataExplorer::QueryController < ::ApplicationController
```

Comment:
I think the plugin is only accessible to admin so we need proper access control here.

Reply:
Ahhh looks like there is an admin constraint on the route so this is OK.